bin/rialto/node/src/overseer.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This is almost 1:1 copy of `node/service/src/overseer.rs` file from Polkadot repository.
+//! The only exception is that we don't support db upgrades => no `upgrade.rs` module.
```

Comment:
CC @rphmeier @ordian any chance we could split `polkadot-service` (or add a feature flag) so that we can avoid building `polkadot-runtime`, but still reference these files?

Reply:
You mean do something like this: https://github.com/paritytech/polkadot/pull/3189#discussion_r646555722?

Reply:
Well, kind of, yeah :) We just don't want a compilation error in case no runtime is selected. @bkchr would it be fine with you to hide `polkadot_runtime` under a feature?

Like maybe we can have compilation error, UNLESS additional `no_runtime` feature is enabled to make sure it's an explicit decision?

Reply:
What has the `polkadot_runtime` to do with the comment above?

Reply:
What is rialto? Another relay chain?

Reply:
> What has the polkadot_runtime to do with the comment above?

This PR attempts to use (generic) parts of `polkadot-service` but with a custom runtime. Currently the code is copied over to **avoid compiling** the entire `polkadot_runtime`. Having a feature in `polkadot-service` to disable the runtime is considered as an alternative approach to avoid copy-pasting the code. In other comment I'm also wondering if it's possible to use cargo `patch` instead to plug a custom runtime and use the entire `polkadot` node instead.

> What is rialto? Another relay chain?

Rialto is bridges-team-internal network. Currently it's just a standalone chain, but since we want to test having the bridge on a parachain, we are converting it into a relay chain.

Reply:
I think we could probably make the polkadot runtime optional as well, as long as we enable it by default.

Reply:
@bkchr if https://github.com/paritytech/polkadot/compare/ao-optional-polkadot-runtime is what you have in mind, I can open a PR :)

Reply:
@svyatonik can you check if https://github.com/paritytech/polkadot/pull/3820 solves the code duplication issue?